app/org/sagebionetworks/bridge/models/accounts/Account.java

```diff
@@ -34,14 +40,90 @@ default ConsentSignature getActiveConsentSignature(SubpopulationGuid subpopGuid)
         return null;
     }
     
+    /**
+     * Gets an immutable copy of the consents this account has for the given subpopulation. Consents should be returned
+     * in the order they were signed (earliest first). Returns an empty list if there are no consents.
+     */
+    default List<ConsentSignature> getConsentSignatureHistory(SubpopulationGuid subpopGuid) {
+        List<ConsentSignature> signatures = Lists.newArrayList();
+        for (Map.Entry<HibernateAccountConsentKey, HibernateAccountConsent> entry : getConsents().entrySet()) {
```

Comment:
Similarly, unit tests for get/setConsentSignatureHistory() should be moved from DAO to AccountTest. Similarly for getAllConsentSignatureHistories().

Reply:
I added tests specifically to verify that these methods properly transform the consents map on Account.